File: cs/Minsk/CodeAnalysis/Syntax/LiteralExpressionSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public sealed class LiteralExpressionSyntax : ExpressionSyntax
{
    public SyntaxToken LiteralToken { get; }
    public object? Value { get; }

    public override SyntaxKind Kind => SyntaxKind.LiteralExpression;
    public override IEnumerable<SyntaxNode> Children => new[] { LiteralToken };

    public LiteralExpressionSyntax(SyntaxToken literalToken) : this(literalToken, literalToken.Value)
    {
    }

    public LiteralExpressionSyntax(SyntaxToken literalToken, object? value)
    {
        LiteralToken = literalToken;
        Value = value;
    }
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/LiteralExpressionSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class LiteralExpressionSyntax(val literalToken: SyntaxToken) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.LiteralExpression
    override val children: List<SyntaxNode>
        get() = listOf(literalToken)
}
